### ISSUE TITLE
Coupons not working for logged out users when `perUserLimit` set to 0

### ISSUE THREAD

**Issue by sjcallender** [craftcms/commerce#1187]
### Description
Previously working coupon codes with no conditions are no longer applying to carts for logged out users as they used to on version 2.2.8. Starting in 2.2.9, there was a PHP error. That was fixed in 2.2.12, but the underlying issue of not applying the discount remain.s

**Logged out, v 2.2.8:**
<img src="https://user-images.githubusercontent.com/984311/71199542-26b9b480-2264-11ea-897f-0f5d0f0c43ca.png" width="320px">

**Logged out, v 2.2.12:**
<img src="https://user-images.githubusercontent.com/984311/71199515-1b668900-2264-11ea-8240-a99d5f3a9630.png" width="320px">

### Steps to reproduce

1. Create a coupon with no conditions. Example: https://share.getcloudapp.com/yAuLlv0k
2. Apply coupon to cart and see that no discounts are applied.

### Additional info

- Craft version: 3.3.18.2
- PHP version: 7.3 and 7.2
- Database driver & version: mysql 5.7
- Plugins & versions: 
```
Amazon S3   1.2.5
Authorize.net for Craft Commerce  1.0.8
Bulk Discount   2.0.0
Bundle Mate   1.0.0
Bundles   1.0.14
Commerce Bulk Pricing   1.1.0
Commerce Insights Components  1.0.0
Craft Commerce  2.2.12
Element API   2.6.0
Freeform  3.5.8
Imager  v2.3.0
Incognito Field   1.1.1.1
Inlin   2.1.1
Inventory   1.0.0
Klaviyo Connect   3.1.2
Letter Price  2.0.0
Linkit  1.1.11
Minify  1.2.9
Navigation  1.2.0
PayPal for Craft Commerce   2.1.0.1
Preparse Field  v1.1.0
Redactor  2.4.0
Retour  3.1.31
Scout   2.1.3
SEOmatic  3.2.37
Super Table   2.3.1
```

**Comment by lukeholder** [comment]
@sjcallender can you see if this is fixed in 2.2.13?

**Comment by sjcallender** [comment]
@lukeholder Yes, this is fixed with 2.2.13. Thanks!!
